avcodec/avcodec.go

```diff
@@ -450,10 +383,35 @@ func (pkt *Packet) GetData() []byte {
 	return data
 }
 
+func (pkt *Packet) GetDataAt(index int) byte {
+	return byte(C.go_get_data_at((*C.uint8_t)(pkt.Data()), C.int(index)))
+}
+
 func (pkt *Packet) SetData(data unsafe.Pointer) {
 	pkt.CAVPacket.data = (*C.uint8_t)(data)
 }
 
+func (pkt *Packet) SetBytes(b []byte) error {
+	size := C.int(len(b))
+	if pkt.CAVPacket.data == nil {
+		ret := C.av_new_packet(pkt.CAVPacket, C.int(len(b)))
+		if ret < 0 {
+			return avutil.NewErrorFromCode(avutil.ErrorCode(ret))
+		}
+	}
+	if pkt.CAVPacket.size < size {
+		ret := C.av_grow_packet(pkt.CAVPacket, size-pkt.CAVPacket.size)
+		if ret < 0 {
+			return avutil.NewErrorFromCode(avutil.ErrorCode(ret))
+		}
+	}
+	for i := 0; i < len(b); i++ {
+		C.set_data_at(pkt.CAVPacket.data, C.int(i), (C.uint8_t)(b[i]))
+	}
```

Comment:
Having spent a bit of time working in C now, we can probably remove the need for this loop if we use `memcpy` in the C function 😅  should be some examples of this in the seekable FIFO test helpers in the DCA.

Reply:
But how do you then copy the []byte go type into a []C.uint8_t c type?